src/components/login/MagicLink.tsx

```diff
@@ -76,45 +76,57 @@ const MagicLink = ({ grantToken, hideCodeInput }: Props) => {
     const requestUiSchema = useConstant(() =>
         custom_generateDefaultUISchema(requestSchema)
     );
-    custom_generateDefaultUISchema;
+
+    const redirectPath = useMemo(
+        () =>
+            grantToken
+                ? `${redirectTo}?${GlobalSearchParams.GRANT_TOKEN}=${grantToken}`
+                : redirectTo,
+        [grantToken, redirectTo]
+    );
+
+    const magicLinkOnSubmitWithToken = useCallback(
+        (formData: { email: string; token: string }) => {
+            return supabaseClient.auth.verifyOTP(
+                {
+                    email: formData.email,
+                    token: formData.token,
+                    type: 'magiclink',
+                },
+                {
+                    redirectTo: redirectPath,
+                }
+            );
+        },
+        [redirectPath, supabaseClient.auth]
+    );
+
+    const magicLinkOnSubmitWithoutToken = useCallback(
+        (formData: { email: string }) => {
+            return supabaseClient.auth.signIn(
+                {
+                    email: formData.email,
+                },
+                {
+                    redirectTo: redirectPath,
+                    shouldCreateUser: loginSettings.enableEmailRegister,
+                }
+            );
+        },
+        [redirectPath, supabaseClient.auth]
+    );
```

Comment:
Moved some of this stuff into `memo`/`callbacks` to make sure they use the correct variable values. Just being a bit safer.